src/ontodia/widgets/classTree.tsx

```diff
@@ -1,66 +1,41 @@
 import * as React from 'react';
 import * as _ from 'lodash';
 
-import { Dictionary, LocalizedString } from '../data/model';
 import { FatClassModel } from '../diagram/elements';
 import { DiagramView } from '../diagram/view';
 import { EventObserver } from '../viewUtils/events';
 import { formatLocalizedLabel } from '../diagram/model';
-
-// bundling jstree to solve issues with multiple jquery packages,
-// when jstree sets itself as plugin to wrong version of jquery
-const jstreeJQuery = require<JQueryStatic>('exports?require("jquery")!jstree');
-require('jstree/dist/themes/default/style.css');
+import { TreeNodes } from './treeNodes';
 
 export interface ClassTreeProps {
     view: DiagramView;
     onClassSelected: (classId: string) => void;
 }
 
-interface ClassTreeElement {
-    id: string;
-    label: ReadonlyArray<LocalizedString>;
-    count: number;
-    children: ClassTreeElement[];
-    a_attr?: {
-        href: string;
-        draggable: boolean;
-    };
-    text?: string;
-    type?: string;
+export interface ClassTreeState {
+    roots?: ReadonlyArray<FatClassModel> | undefined;
+    resultIds?: Array<FatClassModel> | undefined;
+    lang?: Readonly<string> | undefined;
+    searchString?: string | undefined;
 }
 
 const CLASS_NAME = 'ontodia-class-tree';
 
-export class ClassTree extends React.Component<ClassTreeProps, {}> {
+export class ClassTree extends React.PureComponent<ClassTreeProps, ClassTreeState> {
     private readonly listener = new EventObserver();
 
-    private treeRoot: HTMLElement;
-    private jsTree: JQuery;
+    constructor(props: ClassTreeProps) {
+        super(props);
+        this.state = ({ roots: undefined });
+        this.search = _.debounce(this.search, 800 /* ms */);
```

Comment:
After initialization `private readonly searchDebounce = new Debouncer();` idk how to call search soomething like this
`private onSearchKeyup = (e: React.KeyboardEvent<HTMLInputElement>) => {
        this.searchDebounce.call(this.search(e.currentTarget.value)); 
    }`

Can you help me?

Reply:
It should look similar to this:
```
private readonly searchDebounce = new Debouncer(800 /* ms */);

private onSearchKeyup = (e: React.KeyboardEvent<HTMLInputElement>) => { 
    this.searchDebounce.call(() => this.search(e.currentTarget.value));
}
```

Reply:
Like that. Only I had to store the value of a string in a variable, otherwise the search was called with undefined
`const keyValue = e.currentTarget.value;
 this.searchDebouncer.call(() => this.search(keyValue));`

Reply:
Good catch, this happens because React uses [proxy object for events](https://reactjs.org/docs/events.html):
> If you want to access the event properties in an asynchronous way, you should call event.persist() on the event, which will remove the synthetic event from the pool and allow references to the event to be retained by user code.